src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -192,101 +174,25 @@ export function uiRenderMixin(kbnServer, server, config) {
-      basePath,
-      uiSettings,
-    };
-  }
-
   async function renderApp({
-    app,
+    app = { getId: () => 'core' },
     h,
     includeUserProvidedConfig = true,
     injectedVarsOverrides = {},
   }) {
-    const request = h.request;
-    const basePath = request.getBasePath();
-    const uiSettings = await getUiSettings({ request, includeUserProvidedConfig });
-    app = app || { getId: () => 'core' };
-
-    const legacyMetadata = getLegacyKibanaPayload({
-      app,
-      basePath,
-      uiSettings,
-    });
-
-    // Get the list of new platform plugins.
-    // Convert the Map into an array of objects so it is JSON serializable and order is preserved.
-    const uiPluginConfigs = kbnServer.newPlatform.__internals.uiPlugins.browserConfigs;
-    const uiPlugins = await Promise.all([
-      ...kbnServer.newPlatform.__internals.uiPlugins.public.entries(),
-    ].map(async ([id, plugin]) => {
-      const config$ = uiPluginConfigs.get(id);
-      if (config$) {
-        return { id, plugin, config: await config$.pipe(take(1)).toPromise() };
-      } else {
-        return { id, plugin, config: {} };
-      }
-    }));
-
-    const response = h.view('ui_app', {
-      strictCsp: config.get('csp.strict'),
-      uiPublicUrl: `${basePath}/ui`,
-      bootstrapScriptUrl: `${basePath}/bundles/app/${app.getId()}/bootstrap.js`,
-      i18n: (id, options) => i18n.translate(id, options),
-      locale: i18n.getLocale(),
-      darkMode: get(uiSettings.user, ['theme:darkMode', 'userValue'], false),
-
-      injectedMetadata: {
-        version: kbnServer.version,
-        buildNumber: config.get('pkg.buildNum'),
-        branch: config.get('pkg.branch'),
-        basePath,
-        env: kbnServer.newPlatform.env,
-        legacyMode: app.getId() !== 'core',
-        i18n: {
-          translationsUrl: `${basePath}/translations/${i18n.getLocale()}.json`,
-        },
-        csp: {
-          warnLegacyBrowsers: config.get('csp.warnLegacyBrowsers'),
-        },
-        vars: await replaceInjectedVars(
-          request,
-          mergeVariables(
-            injectedVarsOverrides,
-            app ? await server.getInjectedUiAppVars(app.getId()) : {},
-            defaultInjectedVars
-          )
-        ),
-
-        uiPlugins,
-
-        legacyMetadata,
-      },
+    const { http } = kbnServer.newPlatform.setup.core;
+    const { rendering } = kbnServer.newPlatform.__internals;
+    const provider = rendering.getRenderingProvider({
+      request: h.request,
+      uiSettings: h.request.getUiSettingsService(),
+      injectedVarsOverrides,
     });
+    const content = await provider.render(app.getId(), { includeUserSettings: includeUserProvidedConfig });
 
-    const csp = createCSPRuleString(config.get('csp.rules'));
-    response.header('content-security-policy', csp);
-
-    return response;
+    return h
+      .response(content)
+      .type('text/html')
+      .header('content-security-policy', http.csp.directives);
```

Comment:
We're no longer using the `createCSPRuleString` function or similar, so this does not appear to ever do anything except the defaults unless the `server.csp.directives` config is specified (which did not exist prior to this PR). This breaks the ability to set the `csp.rules` config.

It seems like the HttpService could calculate this string once based on the config of `csp.rules` and just expose that instead of adding a new config key.